tests/test_http_response.py

```diff
@@ -669,6 +669,21 @@ def test_body_as_unicode_deprecation_warning(self):
             self.assertEqual(len(warnings), 1)
             self.assertEqual(warnings[0].category, ScrapyDeprecationWarning)
 
+    def test_json_response(self):
+        json_body = b"""{"ip": "109.187.217.200"}"""
+        json_response = self.response_class("http://www.example.com", body=json_body)
+        self.assertEqual(json_response.json(), {'ip': '109.187.217.200'})
+
+        json_body = b"""null"""
+        json_response = self.response_class("http://www.example.com", body=json_body)
+        _NONE = object()
+        self.assertNotEqual(json_response.json(), _NONE)
```

Comment:
I think this would pass even with the old implementation (`_NONE` here is not the same as `_NONE` in the implementation).

In any case, I believe what we need to test is that `json.loads` is only called once when you call `json_response.json()` twice. We probably need to use `mock` here.

Reply:
@Gallaecio great suggestion, thank you! Added mock test. Please review.